lightly/loss/directclr_loss.py

```diff
@@ -0,0 +1,46 @@
+from cProfile import label
+import torch
+import torch.nn as nn
+
+#Adapted from https://github.com/facebookresearch/directclr/blob/main/directclr/main.py
+class InfoNCELoss(nn.Module):
+    """Implementation of InfoNCELoss as required for DIRECTCLR"""
+    def __init__(self, dim:int ,temprature:float = 0.1):
+        """Parameters
+        Args:
+            dim : Dimension of subvector to be used to compute InfoNCELoss.
+            temprature: The value used to scale logits.
+        """
+        self.temprature = temprature
+        #dimension of subvector sent to infoNCE
+        self.dim = dim
+    
+    def normalize(self, x:torch.Tensor) -> torch.Tensor:
```

Comment:
I would raise the question if it's necessary to put this in its own function.

Reply:
Well technically not but I would avoid writing ```torch.nn.functional(x, dim = 1)``` again and again :)